### ISSUE TITLE
Support reading password from file

### ISSUE THREAD

**Issue by septatrix** [Romern/syncMyMoodle#69]
This is necessary to support e.g. dockers and podmans secrets mechanism or the systemd `LoadCredential` directive.
For this to work a config option (or flag) would need to be added which gets a path from which the password can be read.

**Comment by n0toose** [comment]
Should the option be shown to the user or be "internal", as to not encourage anyone to leave their RWTH SSO password laying around in plain text without some sort of a good justification?

**Comment by septatrix** [comment]
It can be shown to the user. Currently one can also put the password in the config file so in that regard there is not too much of a difference

**Comment by n0toose** [comment]
Yeah, completely forgot about that part. Hey, maybe we could use the config file for that?

**Comment by n0toose** [comment]
> This is necessary to support e.g. dockers and podmans secrets mechanism

I am a bit uneducated on the subject, is an `entrypoint.sh` file with

```
#!/bin/bash
BASEDIR=$(dirname "$0");
cd $BASEDIR;

if [[ -n "$PASS" ]];
then
  if [[ -n "$USER" ]];
  then
	  python3 -m syncMyMoodle $TOKEN
  else
	  echo 'You must specify environment variable $USER'
  fi
else
  echo 'You must specify environment variable $PASS'
fi
fi
```

not good enough? (P.S. I wrote the file in bulk)

**Comment by septatrix** [comment]
Not really because there are often ways in which environment variables can be accessed or are leaked.
Docker, Kubernetes, Systemd etc. therefore write the content to files and pass the files as environment variables.
These files must the be read to get the secrets

**Comment by D-VR** [comment]
I think syncMyMoodle should go even further and remove the CLI options to pass user secrets, as that ends up in shell history + process lists.
Should probably also remove having secrets in the config file, as we shouldn't be mixing configuration with secrets imo (and we shouldn't be encouraging users to leave files with passwords around).

My idea would be this:
Only support passing secrets by a file, secret store (keyring, or password manager), or as a interactive one time login which stores just tokens / sets up keyring. This should encourage good security behavior and decrease user friction.

Also touches #63 and #159 

A prerequisite for this work is doing #32 by refactoring the config system to make it easier to work with (both for users as well as internally), before completely revamping how auth works
